src/main/java/com/example/learninghub/judge/JudgeQueue.java

```diff
@@ -35,35 +35,49 @@ public void enqueue(JudgeParams judgeParams, Integer submitID) {
             } catch (InterruptedException e) {
                 throw new RuntimeException(e);
             }
+
             String input = testService.getTests(judgeParams.getProblemId()).get(judgeParams.getTestCase() - 1).getInput();
             String output = testService.getTests(judgeParams.getProblemId()).get(judgeParams.getTestCase() - 1).getOutput();
+            final String[] userOutput = {null};
```

Comment:
Is there a reason why userOutput is an String Array?

Reply:
Indeed, there is, only `final` objects can figure as `lvalue` inside new thread definition. Furthermore, it cannot be `string` since one could not assign any value to it. Hence, in my humble opinion, wracking that object up with an `array` is a clear and reasonable solution to overcome those limitations.